DSharpPlus/Entities/AuditLogs/AuditLogParser.cs

```diff
@@ -35,77 +13,66 @@
-            }
-
-            client.UpdateUserCache(discordUser);
-        }
-
-        //get unique webhooks, scheduledEvents, threads
+        _guild ??= await _client.Cache.TryGetGuildAsync(this._guildId);
+        
+        IEnumerable<DiscordUser> users = auditLog.Users.ToList();
         IEnumerable<DiscordWebhook> uniqueWebhooks = auditLog.Webhooks;
         IEnumerable<DiscordScheduledGuildEvent> uniqueScheduledEvents = auditLog.Events;
-        IEnumerable<DiscordThreadChannel> uniqueThreads = auditLog.Threads;
-        IDictionary<ulong, DiscordWebhook> webhooks = uniqueWebhooks.ToDictionary(x => x.Id);
-
-        //update event cache and create a dictionary for it
-        foreach (DiscordScheduledGuildEvent discordEvent in uniqueScheduledEvents)
+        IEnumerable<DiscordThreadChannel> uniqueThreads = auditLog.Threads; 
+        
+        List<DiscordMember>? discordMembers = new List<DiscordMember>();
+        foreach (DiscordUser user in users)
         {
-            if (guild._scheduledEvents.ContainsKey(discordEvent.Id))
-            {
-                continue;
-            }
-            guild._scheduledEvents[discordEvent.Id] = discordEvent;
-        }
-        IDictionary<ulong, DiscordScheduledGuildEvent> events = guild._scheduledEvents;
+            DiscordMember? cachedMember = await this._client.Cache.TryGetMemberAsync(user.Id, _guildId);
 
-        foreach (DiscordThreadChannel thread in uniqueThreads)
-        {
-            if (guild._threads.ContainsKey(thread.Id))
+            if (cachedMember is null && _guild is not null)
             {
-                continue;
+                cachedMember = _guild._members.TryGetValue(user.Id, out cachedMember) ? cachedMember : null;
             }
-            guild._threads[thread.Id] = thread;
+            
+            cachedMember ??= new DiscordMember
+            {
+                Id = user.Id,
+                Discord = this._client,
+                _guild_id = _guild.Id
+            };
+            
+            discordMembers.Add(cachedMember);
         }
-        IDictionary<ulong, DiscordThreadChannel> threads = guild._threads;
-
-
-        IEnumerable<DiscordMember>? discordMembers = users?
-            .Select(xau => guild._members != null && guild._members.TryGetValue(xau.Id, out DiscordMember? member)
-                ? member
-                : new DiscordMember { Discord = guild.Discord, Id = xau.Id, _guild_id = guild.Id });
-
-        Dictionary<ulong, DiscordMember>? members = discordMembers?.ToDictionary(xm => xm.Id, xm => xm);
-
+        
+        IDictionary<ulong, DiscordScheduledGuildEvent>? events = uniqueScheduledEvents.ToDictionary(x => x.Id);
+        IDictionary<ulong, DiscordThreadChannel>? threads = uniqueThreads.ToDictionary(x => x.Id);
+        IDictionary<ulong, DiscordWebhook> webhooks = uniqueWebhooks.ToDictionary(x => x.Id);
+        IDictionary<ulong, DiscordMember>? members = discordMembers.ToDictionary(x => x.Id);
```

Comment:
nit: these could really just be concrete Dictionaries, but it doesn't really matter